### PR TITLE
Log on ground to plank

### LINKED ISSUE #11649: Add ability to use wood axe/saw to cut up logs on the ground
Currently if you want to cut a log into planks (i.e. two-by-fours and a lot of wood splinters) you need to have the log in your inventory. This means that cutting logs up becomes tedious, as they are both heavy and voluminous, so it becomes a game of pickup and drop, and then going through a bunch of menus.

I think you should either be able to (a)ctivate the axe/saw on items on the tile beneath you, or (even better) add it as a crafting recipe so you can do batch work as well. Currently I find it easier just to cut down another tree and convert that into planks (instead of logs), which is counter intuitive.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9665482-add-ability-to-use-wood-axe-saw-to-cut-up-logs-on-the-ground?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by NickTaylor0**
This resolves #11649 

This allows the player to cut up a log on the tile beneath them (with a confirmation) into planks (2x4s). The other two commits fix some issues with the way that the construction way of converting a trunk to planks works, but if it should be in another PR (or this is an intended way to do this), I can remove it.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
You can skip that "Do you want to use that log?" query. It's not like anyone will miss an accidentally cut-up log. Having to constantly alternate between 'A'->'y' could get annoying, like it was with rubble, foraging and opening crates.

**Comment by NickTaylor-** [comment]
I more had it so that if you are standing on a log, but want to cut up a log into your inventory, you could do it. I can remove it though

**Comment by KA101** [comment]
Meh, the log might be handy for a log wall.  Coolthulhu's been making a pest of xemself for a while now.

**Comment by NickTaylor-** [comment]
Yes, but if you need X logs and Y planks, if you do it the correct number of times, you'll still have the right amount, you might just have them in the wrong places.

**Comment by Coolthulhu** [comment]
That "xem" offended me more than the "pest" part.

If you have one log, you have a way to make more in huge bulk in 20 minutes in-game time. Accidentally sawing a log would require some seriously weird setup.

**Comment by KA101** [comment]
Very well, I regret that in not misgendering you i caused offense, and further that I allowed my temper to show.  What pronoun would you prefer?

As for the topic, forests are common but there are situations where chopping a tree is not advisable.

**Comment by NickTaylor-** [comment]
Yes, but grabbing the log from your inventory vs. the ground shouldn't effectively make a difference, as if you really need that extra log in your inventory on the ground, you can just drop it.

**Comment by Coolthulhu** [comment]
First off I'd prefer not to be called a pest for daring to disagree about game design. We all have our shit days and I see why would you be angry with all that forum drama, but that's not an excuse to just insult someone. I'm trying my best to limit myself to criticism and not "banter".
Direct confrontation is better than passive aggressiveness - if you have something against me, send me a PM.
I wasn't serious about the pronouns. "He" or "they" are both fine. Like someone already mentioned (on wiki?): "xe" is more likely to be offensive than "they", especially now that it is actually used as an insult (with the same meaning as "special snowflake") in some parts of the Internet.

Back to the topic:
The query only appeared while standing directly over the log. Out of all the accidents that could happen when misusing the use menu, this would be one of the least consequential (compare with using meth or drinking bleach).
Wood saw can be used to chop sticks into planks, but most players who would want to do that have no access to logs (which are a much better source of planks).

**Comment by KA101** [comment]
For the sidetrack: So noted.  Try *shit weeks*.

I'll consider this one.

File: src/iuse.cpp [hunk 1]
```diff
@@ -6032,15 +6032,36 @@ int iuse::cut_log_into_planks(player *p, item *it)
...
-        return itm.type->id == "log";
-    } );
-    item *cut = &( p->i_at( pos ) );
+    item* cut = nullptr;
+
+    // Check if player is standing on any lumber
+    for (auto &i : g->m.i_at(p->posx(), p->posy())) {
+        if (i.type->id == "log")
+        {
+            cut = &i;
```
Comment (BevapDin) [inline src/iuse.cpp]: Storing the pointer in `cut` is pointless, just remove the item:
```C++
g->m.i_rem( ..., &i )
```

`cut` is not needed at all until it is used to store the result of `g->inv_for_filter` (like it was done before the changes).

Below the test for `cut == nullptr` will *always* be true: the only place before that where `cut` has been set to something, is right before the function is left with the `return`. Even if the function was not left, `cut` will point to an non-existing thing, the item it pointed to was removed via `g->m.remove_item`.

About `cut_log_into_planks`:
I recommend removing the `it` parameter completely and return nothing at all. The caller does either not care about the charges that was returned (for constructions) or the caller already has the item and can get the charges from it. This avoids the usage of the null pointer.

**Comment by NickTaylor-** [comment]
I completely agree with your first points, just design remnants from an early version I had working.

Your second point however, `cut_log_into_planks` needs `it` and a return for when you cut up a log using the activate function of a chainsaw. This way it actually uses up a charge when you cut up the log.

**Comment by Coolthulhu** [comment]
Maybe alter it so that the internal function only returns a bool (was the log cut) and the external one returns charges based on that. The invocation in construction.cpp drops the return value anyway, so it's not like anything would get worse here.

**Comment by NickTaylor-** [comment]
That does make more sense

**Pull request merged**